api/graphql/reservation_units/reservation_unit_serializers.py

```diff
@@ -419,17 +490,131 @@ def validate_for_publish(self, data):
                 "minPersons can't be more than maxPersons"
             )
 
+        if not data.get("pricings", []):
+            raise serializers.ValidationError(
+                "pricings is required and must have one ACTIVE and one optional FUTURE pricing"
+            )
+
+    def validate_pricing_fields(self, data):
+        pricings = data.get("pricings", [])
+        if not pricings:
+            return
+
+        active_count = 0
+        future_count = 0
+        other_count = 0
+
+        for pricing in pricings:
+            if ReservationUnitPricingHelper.is_active(pricing):
+                active_count += 1
+                if pricing.get("begins") > date.today():
+                    raise serializers.ValidationError(
+                        "ACTIVE pricing must be in the past or today"
+                    )
+            elif ReservationUnitPricingHelper.is_future(pricing):
+                future_count += 1
+                if pricing.get("begins") <= date.today():
+                    raise serializers.ValidationError(
+                        "FUTURE pricing must be in the future"
+                    )
+            else:
+                other_count += 1
+
+        is_draft = data.get("is_draft", getattr(self.instance, "is_draft", False))
+        if (not is_draft and active_count != 1) or (is_draft and active_count > 1):
```

Comment:
Would it be sufficient to just check that the active_count is not zero? Can't figure out why we need multiple active pricings even when in draft state

So disregard the `is_draft` here and just iffing => `if active_count < 1: <kaboom>`

Reply:
We can't do that because pricings are not required in the draft state. So when reservation unit is draft, `active_count` can be zero.